core/vm/contracts_state.go

```diff
@@ -150,3 +164,29 @@ func (c *stakingPrecompile) RunWriteCapable(
 	//}
 	return nil, errors.New("[StakingPrecompile] Received incompatible stakeMsg from staking.ParseStakeMsg")
 }
+
+type roStakingPrecompile struct{}
+
+func (c *roStakingPrecompile) RequiredGas(
+	evm *EVM,
+	contract *Contract,
+	input []byte,
+	readOnly bool,
+) (uint64, error) {
+	return GasQuickStep, nil
```

Comment:
The gas seems too small?

Reply:
This one was me too. What would you rather charge, same as the write version of the staking precompile (21000 + data cost)?